LibertyDSNP/example-client
# add ability to follow and unfollow users

Purpose
---------------
Users need a way to follow and unfollow each other.
https://www.pivotaltracker.com/story/show/177348398

Solution
---------------
This PR adds the ability to follow and unfollow users that are following the current user and the ability to unfollow users the logged in user is currently following. It lays the grounds work for following arbitrary users, but we are waiting for #79 for the UI to implement that functionality.

Change summary
---------------
* Add sdk methods to follow and unfollow.
* Modify ConnectionListProfiles to call those methods when the existing buttons are pressed.
* Modify the graphSlice in redux to store a RelationshipStatus enum instead of a boolean, so we can track when a following/followed by relationship has been updated by the user, but not yet recorded on chain.
* Add styling to the graph relationship buttons for the updating state.

Steps to Verify
----------------
1. Start the app in a clean populated state.
2. Open the "followers" tab in the profile block
3. Click one of the "follow" buttons for a user that is not yet followed and approve the transaction.
4. Note that the button you clicked is now in the "updating state.
5. Open the "following" tab and note that the new follow has been added (probably still in the updating state).
6. Note that you see posts from the new follow in the "My Feed" section.
7. Wait for the transaction to go through.
8. Note that the updating button is replaced with an "unfollow" button in the "Followers" and "Following" tabs. 
9. Reload the page, and note that all the changes above still exist once the page is fully loaded.
10. In the "following" tab, choose a user that is also in the "followers" tab, click the "unfollow" button and approve the transaction.
11. Note that the buttons for this user in both tabs are now in the updating state (but nothing has changed otherwise).
12. Wait for the transaction to go through.
13. Note that this user is now removed from the "following" tab, and has a "follow" button in the "followers" tab.
14. Note that you can no longer see this person's posts in the "My Feed" section.
15. Reload the page to demonstrate these changes are persisted.
